my-app/src/App.jsx

```diff
@@ -88,24 +84,76 @@ export default class App extends React.Component {
     ));
 
   renderCart = () =>
-    this.state.cartList.map(cartItem => (
+    this.state.cartList.map((cartItem, index) => (
       <CartItem
+        remove={this.removeFromCart}
+        decrease={this.decrease}
+        increase={this.addToCart}
+        key={`${cartItem.id}-${index}`}
+        itemId={cartItem.id} // id-ul unic
         peak={cartItem.peak}
         county={cartItem.county}
         cottage={cartItem.cottage}
+        countItem={cartItem.countItem} // nr. de trips
       />
     ));
+  cleanCart = () => {
+    this.setState({ cartList: [] });
+  };
+
+  addToCart = itemId => {
+    // debugger;
+    const itemToAdd = this.state.mountains.find(item => item.id === itemId);
+    if (!itemToAdd) {
+      return undefined;
+    }
+
+    const cartUpdated = this.state.cartList.slice();
+    const cartItem = cartUpdated.find(item => item.id === itemId);
+    if (!cartItem) {
+      // create a new trip
+      const newItem = Object.assign({}, itemToAdd);
+      newItem.countItem = 1;
+      cartUpdated.push(newItem);
+    } else {
+      // increse the trips
+      cartItem.countItem++;
+    }
+    this.setState({ cartList: cartUpdated });
+  };
+  decrease = itemId => {
+    debugger;
```

Comment:
sterge debugger-ul ;) daca te obisnuiesti de acum sa nu lasi "console.log" sau "debugger", cand vei ajunge sa scri cod in productie, acest lucru nu va mai fi o problema :D